cunumeric/module.py

```diff
@@ -1460,17 +1460,30 @@ def check_shape_dtype(
             f"All arguments to {func_name} "
             "must have the same number of dimensions"
         )
-    if ndim > 1 and _builtin_any(
-        shape[1:axis] != inp.shape[1:axis]
-        and shape[axis + 1 :] != inp.shape[axis + 1 :]
-        for inp in inputs
-    ):
-        raise ValueError(
-            f"All arguments to {func_name} "
-            "must have the same "
-            "dimension size in all dimensions "
-            "except the target axis"
-        )
+
+    if func_name == "stack":
+        shapes = {inp.shape for inp in inputs}
+        if len(shapes) != 1:
+            raise ValueError(
+                f"all input arrays must have the same shape for {func_name}"
+            )
+    else:
+        if ndim == 1 and func_name == "hstack":
+            axis = 0
+        if ndim >= 1:
+            if axis < 0 and axis >= -ndim:
+                axis += ndim
+            if _builtin_any(
+                shape[:axis] != inp.shape[:axis]
+                or shape[axis + 1 :] != inp.shape[axis + 1 :]
```

Comment:
Are we sure about this? I still think comparing arrays together with `or` may be a problem and that the "any" call needs to be distributed over the "or"

https://github.com/nv-legate/cunumeric/pull/828#discussion_r1126837467

Reply:
In this case we're working with ndarray shapes, which are tuples, so IIUC tuple-wide inequality will work with `or` as written. But maybe an alternative formulation would be more clear?

Reply:
@bryevdv Do you suggest the below code is better?
```
if _builtin_any(
    _builtin_any(shape[:axis] != inp.shape[:axis]）
    or 
    _builtin_any(shape[axis + 1 :] != inp.shape[axis + 1 :]）
for inp in inputs)
```

Reply:
@robinw0928 disregard, `shape` values are tuples here, your original code is fine